docs/content_block_manager/adr/0003-render-blocks-at-publish-time.md

```diff
@@ -0,0 +1,120 @@
+# 3. Render blocks at publish time
+
+Date: 2025-05-20
+
+## Context
+
+We are introducing the concept of "views" for entire blocks, in addition to being able to embed individual
+attributes. These views can be ordered in specific ways (for example, a user can choose to show the telephone
+number before the address in the case of a contact), as well as allowing custom views to be created to allow
+only specific attributes to be shown.
+
+At the moment, the order and presentation of blocks is controlled by code in the 
+[`content_block_tools`](https://github.com/alphagov/content_block_tools) gem. Each block type has a presenter
+which returns a representation of all the attributes in a block, styled in a particular way. For example, the
+render method for a contact block looks like this:
+
+```ruby
+def default_content
+  content_tag(:div, class: "contact") do
+    concat content_tag(:p, content_block.title, class: "govuk-body")
+    concat(email_addresses.map { |email_address| email_address_content(email_address) }.join.html_safe) if email_addresses.any?
+    concat(phone_numbers.map { |phone_number| phone_number_content(phone_number) }.join.html_safe) if phone_numbers.any?
+  end
+end
+```
+
+This returns the elements in a strict order, styled in a particular way, and can only be changed by developers.
+
+Additionally, it will be difficult to allow users to create new views, without having some mechanism to tell the
+gem that we expect to see only specific attributes in the view.
+
+## Decision
+
+We will move the logic for rendering specific views away from the Content Block Tools gem to Content Block Manager.
+
+We will then add a separate `views` object to each schema for a content block. At the time of publication, the `views`
+object will be sent to the Publishing API with a HTML representation of the view. For example, in the case of contacts,
+a view could look like this:
+
+```html
+<div class="contact">
+    <p>My contact details</p>
+    <p>Tel: 12345</p>
+    <p>Email: <a href="mailto:abc@example.com">abc@example.com</a></p>
+</div>
+```
+
+This will then be sent in the JSON payload as a default `views` object within the payload sent to Publishing API. An 
+abridged version of the payload could look like this:
+
+```json
```

Comment:
Another thing I’ve just thought about is rather than sending the HTML to Publishing API, we could just send an ordered array as a `blocks` element, with the order of the blocks specified. So for addresses, the default would look like this:

```
{
  "blocks: {
    "default": ["title", "addresses", "email", "telephones"],
  }
}
```

This might be simpler, but would give us less flexibility if we wanted to introduce (for example) custom styling / presentation to blocks, as well as the ability to add arbitrary text around blocks